css/css-typed-om/the-stylepropertymap/properties/logical.html

```diff
@@ -13,15 +13,59 @@
 <script>
 'use strict';
 
+const logical = {
+  sides: [
+    'block-start', 'block-end', 
+    'inline-start', 'inline-end',
+  ],
+  shorthands: [
+    'block', 
+    'inline',
+  ],
+};
+
 for (const prefix of ['margin-', 'inset-', 'padding-']) {
-  for (const suffix of ['block-start', 'block-end', 'inline-start', 'inline-end']) {
-    runPropertyTests(prefix + suffix, [
+  for (const side of logical.sides) {
+    runPropertyTests(prefix + side, [
       // TODO: Test 'auto'
       { syntax: '<percentage>' },
       { syntax: '<length>' },
     ]);
   }
 }
 
-// TODO: test shorthands and other logical properties.
+// BORDERS
+for (const side of [...logical.sides, ...logical.shorthands]) {
+  runPropertyTests('border-' + side, [
+    //{ syntax: 'thin solid green' },
+    //{ syntax: 'thin solid' },
+    //{ syntax: 'thick' },
+    { syntax: 'none' },
+  ]);
+
+  runPropertyTests(`border-${side}-width`, [
+    { syntax: 'thin' },
+    { syntax: 'medium' },
+    { syntax: 'thick' },
+    { syntax: '<length>' },
+  ]);
+
+  runPropertyTests(`border-${side}-color`, [
+    { syntax: 'currentcolor' },
+    //{ syntax: '<color>' },
+  ]);
+
+  runPropertyTests(`border-${side}-style`, [
+    { syntax: 'none' },
+    { syntax: 'solid' },
+  ]);
+}
+
+// border radius
+for (const side of logical.sides) {
+  runPropertyTests(`border-${side}-radius`, [
```

Comment:
This won't work well, border radii refer to box corners instead of sides, so there is e.g. `border-start-start-radius` but no `border-block-start-radius`.

You could use something like

```js
const logical = {
  axes: ['block', 'inline'],
  sides: ['block-start', 'block-end', 'inline-start', 'inline-end'],
  corners: ['start-start', 'start-end', 'end-start', 'end-end'],
};
for (const corner of logical.corners) {
  runPropertyTests(`border-${corner}-radius`, [
```

Reply:
nice, another regression from the refactor. suggestion is excellent, will use 👍